### DISCUSSION TITLE
Get the member which is using the current Attribute

### DISCUSSION THREAD

**Discussion by SlimeNull** [dotnet/csharplang#5017] [Language Ideas]
hope this feature can be supported, this will be very helpful
```csharp
class MyAttribute : Attribute {
    void SomeMethod(){
        MemberInfo member = this.GetUserMember();    // this method should be from class 'Attribute'
    }
}
```

**Comment by svick** [comment]
Can you explain why you think this would be useful?

Usually, attribute classes don't do any any processing at all, they just hold the data. It's the type that uses reflection to retrieve the attribute that does the processing, and it has all the information it needs for that (including the attribute and the member it's applied to). How would your proposal improve such code?

**Reply by SlimeNull** [reply]
If I want to get all members which has 'myattribute', now we have to check all members in the assembly, but if we can get the user in attribute, we can use this:
```csharp
class myattribute : Attribute
{
    public static List<MemberInfo> Users { get; } = new();
    public myattribute ()
    {
        Users.Add(this.GetUser());
    }
}
```

**Reply by alexrp** [reply]
Attributes don't get instantiated until you actually query the attributes on something (`type.GetCustomAttribute<PureAttribute>()`, `method.GetCustomAttributes(true)`, etc...). You seem to be assuming that they're instantiated at startup or something like that, which is not the case.

**Reply by SlimeNull** [reply]
emmm，OMG，I do thought that like what you said, thank you very much!

**Reply by Mrxx99** [reply]
If you think scanning the whole assembly is to slow for you, you could do this with the new C# source generators.
